src/machine.rs

```diff
@@ -404,6 +412,60 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for Evaluator<'tcx> {
         Ok(())
     }
 
+    fn eval_maybe_thread_local_static_const(
+        ecx: &InterpCx<'mir, 'tcx, Self>,
+        mut val: mir::interpret::ConstValue<'tcx>,
+    ) -> InterpResult<'tcx, mir::interpret::ConstValue<'tcx>> {
+        match &mut val {
+            mir::interpret::ConstValue::Scalar(Scalar::Ptr(ptr)) => {
```

Comment:
Hm, seems a bit risky do only do something on a `Scalar` here... in principle there could be thread-locals in any constant. I suppose right now that does not happen, but if/when it does, we'd silently do the wrong thing.

Could you at least add a FIXME for that? I really don't like these incomplete solutions, but also if we really solve https://github.com/rust-lang/rust/issues/70685 soon then this will all change again. I just saw too many "temporary solutions" become permanent...

Reply:
Also please move the body of this function to the thread manager; `machine.rs` should ideally only contain glue code and no non-trivial logic.